src/main/java/zd/zero/waifu/motivator/plugin/service/WaifuGatekeeper.kt

```diff
@@ -0,0 +1,44 @@
+package zd.zero.waifu.motivator.plugin.service
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.ServiceManager
+import zd.zero.waifu.motivator.plugin.settings.PluginSettingsListener
+import zd.zero.waifu.motivator.plugin.settings.WaifuMotivatorPluginState
+import zd.zero.waifu.motivator.plugin.settings.WaifuMotivatorState
+
+class WaifuGatekeeper : Disposable {
+    companion object {
+        val instance: WaifuGatekeeper
+            get() = ServiceManager.getService(WaifuGatekeeper::class.java)
+    }
+
+    private val connection = ApplicationManager.getApplication().messageBus.connect()
+
+    private var allowedWaifu: Set<String> = constructList(WaifuMotivatorPluginState.getPluginState())
+
+    private fun constructList(pluginState: WaifuMotivatorState): Set<String> =
```

Comment:
*side-comment*: I just find it weird that `constructList` returns a `Set`, maybe rename it to something like `constructWaifusFromState` or `createFromState` but it's up to you, It's really hard to name things! >.<

Reply:
Fixed!